### PR TITLE
increased min PHP version to 5.6

### PR DESCRIPTION
**Pull request by vondrasekpavel**
added dev-dependencies (PHPUnit 5.7 & PHPCR-Utils)

removed PHPUnit deprecations

& code cleanup

### CODE REVIEW

**Comment by vondrasekpavel** [comment]
And added hotfix for broken CndParserTest::testScannerErrorComment on PHP7

File: tests/Writing/CopyMethodsTest.php [hunk 1]
```diff
@@ -79,24 +85,6 @@ public function testWorkspaceCopy()
         $this->assertNotEquals($sfile->getPropertyValue('jcr:data'), $dfile->getPropertyValue('jcr:data'));
     }
 
-    public function testCopyPreserveChildOrder()
```
Comment (dbu) [inline tests/Writing/CopyMethodsTest.php]: why do you delete this test?

File: tests/PhpcrUtils/CndParserTest.php [hunk 2]
```diff
@@ -93,22 +98,29 @@ public function testParseError()
...
      */
     public function testScannerErrorComment()
     {
+        /**
+         * Remove this PHP7 only condition after PHPCR-Utils #168
+         * (https://github.com/phpcr/phpcr-utils/issues/168) is closed
+         */
+        if (PHP_MAJOR_VERSION >= 7) {
+            $this->markTestSkipped();
+        }
```
Comment (dbu) [inline tests/PhpcrUtils/CndParserTest.php]: please move this into a separate commit, so that we can easily revert it when we fixed the bug

File: inc/BaseCase.php [hunk 3]
```diff
@@ -11,19 +11,26 @@
...
 use PHPCR\SessionInterface;
 use PHPCR\NodeInterface;
 use DateTime;
+use PHPUnit_Framework_SkippedTestSuiteError;
+use PHPUnit_Framework_TestCase;
+use PHPUnit_Util_Filter;
 
 // PHPUnit 3.4 compat
-if (method_exists('PHPUnit_Util_Filter', 'addDirectoryToFilter')) {
+if (method_exists(PHPUnit_Util_Filter::class, 'addDirectoryToFilter')) {
```
Comment (vondrasekpavel) [inline inc/BaseCase.php]: Ok, good :)

File: composer.json [hunk 4]
```diff
@@ -27,9 +27,13 @@
         }
     ],
     "require": {
-        "php": ">=5.3.0",
+        "php": "^5.6|7.0.x|7.1.x",
```
Comment (vondrasekpavel) [inline composer.json]: Ah, I copied it from phpcr/utils :D Ok, I will update it :)

File: tests/Writing/CopyMethodsTest.php (hunk 1 above)
Comment (vondrasekpavel) [inline tests/Writing/CopyMethodsTest.php]: Ah, I don't understand how I dit it :o I had to be tired :( 

I will revert it. I noticed that this test use php 5.4 short array syntax but was declared 5.3 support...

Comment (vondrasekpavel) [inline tests/Writing/CopyMethodsTest.php]: Ok, we resolved it :)

File: tests/PhpcrUtils/CndParserTest.php (hunk 2 above)
Comment (vondrasekpavel) [inline tests/PhpcrUtils/CndParserTest.php]: Ok, it will be clever :)

**Comment by vondrasekpavel** [comment]
> great work! i commented on a few things. i was about to complain about moving @expectedException to $this->expectException, but then found https://thephp.cc/news/2016/02/questioning-phpunit-best-practices so i guess we should change this.

I moved it from @expectedException to $this->expectException because I wanted to use short class name and it doesn't worked and it was fastest fix which I found.

> i just found that styleci was not enabled for this repository, but we have the styleci.yml file... can you also adjust https://github.com/phpcr/phpcr-api-tests/blob/master/.styleci.yml#L4 please? when you push the update, we should get a styleci build.

Good, I agree. And what do you think about PHP-CS-Fixer?

File: tests/PhpcrUtils/CndWriterTest.php [hunk 5]
```diff
@@ -86,18 +87,14 @@ function ($prefix) {
...
         ;
         $cndWriter = new CndWriter($ns);
-        $res = $cndWriter->writeString(array($tpl));
+        $res = $cndWriter->writeString([$tpl]);
 
         $this->assertEquals($cnd, $res);
     }
 }
-
-abstract class MockNamespaceRegistry implements \Iterator, NamespaceRegistryInterface
```
Comment (vondrasekpavel) [inline tests/PhpcrUtils/CndWriterTest.php]: Doesn't it conflict with PSR-1: "This means each class is in a file by itself, and is in a namespace of at least one level: a top-level vendor name." ?

And in .styleci.yml we declared PSR-2 support and it says: "Code MUST follow a “coding style guide” PSR [PSR-1]."

File: tests/Reading/PropertyReadMethodsTest.php [hunk 6]
```diff
@@ -292,34 +316,39 @@ public function testGetDoubleAndDecimalSame()
...
         }
         //check correct values and sort order
-        $expectedArray = array(
-                new \DateTime('2011-04-22T14:34:20+01:00'),
-                new \DateTime('2011-10-23T14:34:20+01:00'),
-                new \DateTime('2010-10-23T14:34:20+01:00'), );
+        $expectedArray = [
+                new DateTime('2011-04-22T14:34:20+01:00'),
+                new DateTime('2011-10-23T14:34:20+01:00'),
+                new DateTime('2010-10-23T14:34:20+01:00')
```
Comment (vondrasekpavel) [inline tests/Reading/PropertyReadMethodsTest.php]: My fault will fix it. :)

Cool! At this time I don't know nothing about open source, unit tests and OOP... In March 2010 I created my very first project - browser game written in procedural PHP with MySQL it was spaghetti code but it was very fast because I spent long time with SQL optimization and PHP micro optimization it was funny :D :D

**Comment by vondrasekpavel** [comment]
I updated .styleci but Styleci still doesn't like my code :(

File: tests/PhpcrUtils/CndWriterTest.php (hunk 5 above)
Comment (dbu) [inline tests/PhpcrUtils/CndWriterTest.php]: fair enough. lets keep the change.

**Comment by dbu** [comment]
click the "Details" link next to the style-ci line. then login with styleci and you can make it create a pull request against your branch to fix the violations it found. there seems to be a couple of `array` => `[]` that your tool missed, and a few whitespace fixes.

File: tests/Writing/CloneMethodsTest.php [hunk 7]
```diff
@@ -266,33 +275,30 @@ public function testReferenceableDestNodeWithNonReferenceableSourceNode()
         self::$destWs->cloneFrom($this->srcWsName, $srcNode, $dstNode, true);
     }
 
-    /**
-     * @expectedException \PHPCR\NoSuchWorkspaceException
-     */
     public function testCloneNoSuchWorkspace()
     {
+        $this->expectException(NoSuchNodeTypeException::class);
```
Comment (dbu) [inline tests/Writing/CloneMethodsTest.php]: this is the wrong exception

File: tests/Writing/CloneMethodsTest.php [hunk 8]
```diff
@@ -408,41 +413,39 @@ public function testGetCorrespondingNode()
...
 
         $this->assertEquals($srcNode, $clonedNode->getCorrespondingNodePath($this->srcWsName));
     }
 
-    /**
-     * @expectedException \PHPCR\NoSuchWorkspaceException
-     */
     public function testGetCorrespondingNodeNoSuchWorkspace()
     {
+        $this->expectException(NoSuchNodeTypeException::class);
```
Comment (dbu) [inline tests/Writing/CloneMethodsTest.php]: wrong exception

Comment (vondrasekpavel) [inline tests/Writing/CloneMethodsTest.php]: I'm so sorry :(

**Comment by vondrasekpavel** [comment]
snimek obrazovky 2017-01-27 v 14 28 55

Yes I test it in Jackalope-Dbal vendor always :)

**Comment by dbu** [comment]
thanks! squashed on the CLI and merged as 648c8a7b1dee3c16e5cb63c89cd21794064974cf and 6ca175b8048fd492088022cf3e2188236fc34706

tagged as 2.1.16

**Comment by vondrasekpavel** [comment]
It's great we have new version! :) 

Thank you!

**Comment by vondrasekpavel** [comment]
We have wrong php version :(

- phpcr/phpcr-api-tests 2.1.16 requires php ^5.6|7.0.x -> your PHP version (7.1.1) does not satisfy that requirement.

**Comment by dbu** [comment]
fixed in 2.1.17
